azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -689,6 +689,14 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 							Sensitive:    true,
 							ValidateFunc: validation.StringLenBetween(8, 123),
 						},
+						"license": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(containerservice.LicenseTypeNone),
```

Comment:
None shouldn't be supported here - the field should be omitted if it's None (since this is Optional)

```suggestion
```